sdk/core/Azure.Core/src/Shared/DiagnosticScope.cs

```diff
@@ -292,8 +292,11 @@ public void AddLink(string traceparent, string tracestate, IDictionary<string, s
             public Activity? Start()
             {
                 _currentActivity = StartActivitySourceActivity();
-
-                if (_currentActivity == null)
+                if (_currentActivity != null)
+                {
+                    _currentActivity.AddTag("az.schema_url", "https://opentelemetry.io/schemas/1.17.0");
```

Comment:
I suggest bundling this change with mappings and actually complying to otel schema 1.17.0, (we're not compliant now)

Changes needed (https://github.com/Azure/azure-sdk/blob/main/docs/tracing/distributed-tracing-conventions.yml)


HTTP:
- `serviceRequestId` -> `az.service_request_id`
- `clientRequestId` -> `az.client_request_id`
- add `net.peer.name` (endpoint)
- add `net.peer.port` (if not 443), i.e. probably never, but it should be of int type
- http.status_code should be int

Messaging:
- add `messaging.system` - either 'eventhubs' or 'servicebus'
- add [`messaging.operation`](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/messaging.md#operation-names) (one of `publish`, `receive` or `process`)
- `message_bus.destination` -> `messaging.destination.name`
- `peer.address` -> `net.peer.name`

Or, if you prefer to keep PRs small and would be sending others with attribute changes before next core release, we can do this as well :)

Reply:
I've added the HTTP changes in this PR other than enabling suppression by default (but perhaps that is already done?). I will do the messaging changes in a separate PR.

Reply:
Looks like we also need messaging.batch.message_count for batch operations in messaging.

Reply:
Potentially also message.destination.name and message.source.name?
https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/messaging.md#producer-attributes